lib/charms/loki_k8s/v1/loki_push_api.py

```diff
@@ -349,6 +352,72 @@ def _promtail_error(self, event):
     )
 ```
 
+## LogForwarder Library Usage
+
+Let's say that we have a workload charm that produces logs to the standard output (stdout),
+and we need to send those logs to a workload implementing the `loki_push_api` interface,
+such as `Loki` or `Grafana Agent`.
+
+Use the `LogForwarder` class by instantiating it in the `__init__` method of the charmed
+operator. There are two ways to provide Loki endpoint(s) to the forwarder. You can let the
+object extract the endpoint(s) from relation data, or you can explicitly pass the Loki
+endpoint(s) to forward your logs to.
+
+1. If you want your charm to relate to another implementing the `loki_push_api` interface,
+   you only need to instantiate the object; for example:
+
+   ```python
+   from charms.loki_k8s.v1.loki_push_api import LogForwarder
+
+   ...
+
+      def __init__(self, *args):
+          ...
+          self._log_forwarder = LogForwarder(self)
+   ```
+
+   The `LogForwarder` will listen to relation events out-of-the-box and enable or disable
+   the log forwarding accordingly. Next, modify the `metadata.yaml` file to add:
+
+   - The `log-forwarding` relation in the `requires` section:
+     ```yaml
+     requires:
+       log-forwarding:
```

Comment:
in this case, the example you give above will not work, because the LogForwarder's endpoint defaults to 'logging', not 'log-forwarding'.